Facility Home/Overview Page redesign

SiddanthNayak opened the pull request:
## Proposed Changes

Fixes #14322, https://openhealthcarenetwork.atlassian.net/browse/ENG-826
- Redesigned certain aspects of the Home Page

<img width="3837" height="2157" alt="image" src="https://github.com/user-attachments/assets/617f2e52-ffbf-4f35-b55a-e54f69b33e5a" />

Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is placed in I18n files.
- [x] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [x] Request peer reviews.
- [x] Complete QA on mobile devices.
- [x] Complete QA on desktop devices.
- [ ] Add or update Playwright tests for related changes


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

* **New Features**
  * Analytics shortcut now appears when enabled.
  * Personalized time-of-day greeting added.
  * New illustrated icons enhance dashboard shortcuts.

* **Improvements**
  * Welcome header redesigned with a decorative banner.
  * Quick Links now feature richer cards, updated labels, icons, and a four-column layout.
  * Custom dashboard links integrated into the updated card design.

* **Localization**
  * Added English translations for analytics and dashboard labels.
  * Improved Malayalam wording for encounter management.

* **Tests**
  * Added end-to-end coverage for navigation to key pages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SiddanthNayak committed:
homepage redesign

SiddanthNayak committed:
fixed typo and styles

SiddanthNayak committed:
removed role from banner image

SiddanthNayak committed:
added appointment permission and route for appointments shortcut

SiddanthNayak committed:
changed translation key for greeting

vinutv committed:
Make home banner responsive using split images + gradient

SiddanthNayak committed:
style changes

SiddanthNayak committed:
show analytics only if plug is enabled

SiddanthNayak committed:
fixed homepage greeting not formatting prefix and suffix

SiddanthNayak committed:
added variable width for each breakpoint

SiddanthNayak committed:
Merge branch 'develop' into feat/14322/home-page-redesign

SiddanthNayak committed:
Merge branch 'develop' into feat/14322/home-page-redesign

SiddanthNayak committed:
Merge remote-tracking branch 'upstream/develop' into feat/14322/home-page-redesign

nihal467 reviewed (changes requested):
- The PR functionality looks good. Add a playwright test to verify that the redirection via card from the overview page is functional

SiddanthNayak committed:
added playwright tests

SiddanthNayak committed:
Merge branch 'develop' into feat/14322/home-page-redesign

SiddanthNayak committed:
fixed typo

SiddanthNayak commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,41 @@
+
+let facilityId: string;
+
+test.describe("Payment Reconciliation", () => {
```
resolved.

SiddanthNayak committed:
moved getFacilityId inside beforeEach

SiddanthNayak committed:
memoized plugs check

SiddanthNayak commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,41 @@
+    await page.getByRole("link", { name: /services/i }).click();
+
+    await expect(page).toHaveURL(`/facility/${facilityId}/services`);
+  });
```
encounter and appointments usually have extra added filters to the url, so contains is the best case here.

SiddanthNayak commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +50,52 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-  const { canViewSchedule, canListEncounters } = getPermissions(
+  const isAnalyticsEnabled = useCareApps().some(
+    (plugin) => plugin.plugin === "care_analytics_fe",
+  );
```
resolved

SiddanthNayak committed:
Merge remote-tracking branch 'upstream/develop' into feat/14322/home-page-redesign

bodhish commented:
@SiddanthNayak why are we not merging this?

SiddanthNayak committed:
Merge remote-tracking branch 'upstream/develop' into feat/14322/home-page-redesign

SiddanthNayak committed:
Merge branch 'develop' into feat/14322/home-page-redesign

yash-learner committed:
Merge branch 'develop' into feat/14322/home-page-redesign

yash-learner committed:
Make style improvements matching the figma design

yash-learner committed:
Match figma

- Add new icons to the cutomIocns
- Update overview syles to figma and make it repsonsive

yash-learner committed:
Merge branch 'develop' into feat/14322/home-page-redesign

yash-learner committed:
Add i18n keys

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +50,52 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    const hour = new Date().getHours();
+    if (hour < 12) return "Morning";
+    if (hour < 18) return "Afternoon";
+    return "Evening";
```
Fixed it

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+        <div className="relative border-3 border-white overflow-hidden text-gray-900 rounded-xl bg-[linear-gradient(to_right,#F9FAFB_0%,#F0F0E6_60%)] sm:bg-[linear-gradient(to_right,#F9FAFB_0%, #F9FAFB_70%, #F0F0E6_100%)] lg:bg-[linear-gradient(to_right,#F5F8FF_0%,#F9FAFB_50%,#F0F0E6_100%)]">
+          <div
+            className="absolute w-full h-full bg-right bg-no-repeat bg-contain rounded-lg mix-blend-darken -right-8 md:right-0"
+            style={{ backgroundImage: "url('/images/home-banner-icon.webp')" }}
```
All the modern devices should supported its been more than a decade, I think people should not be using that so outdated browsers for critical healthcare apps

<img width="1772" height="1700" alt="image" src="https://github.com/user-attachments/assets/4fdb502a-f4f1-4e2f-91e0-dcdb99d3618d" />


cc: @coderabbitai

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                {t("welcome_back_to_hospital_dashboard")}
-              </p>
-            </div>
+        <div className="relative border-3 border-white overflow-hidden text-gray-900 rounded-xl bg-[linear-gradient(to_right,#F9FAFB_0%,#F0F0E6_60%)] sm:bg-[linear-gradient(to_right,#F9FAFB_0%, #F9FAFB_70%, #F0F0E6_100%)] lg:bg-[linear-gradient(to_right,#F5F8FF_0%,#F9FAFB_50%,#F0F0E6_100%)]">
```
outdated

<img width="830" height="165" alt="image" src="https://github.com/user-attachments/assets/74aca77e-150f-4c4f-b0e1-709eb3af47f6" />

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+          />
+
+          <div className="relative z-10 p-4 md:p-8 max-w-[250px] md:max-w-[280px] lg:max-w-lg">
+            <h1 className="mb-2 font-bold text-gray-900 text-2xl wrap-break-word">
```
dude we are v4 so this is valid

yash-learner commented on public/locale/en.json:
```diff
@@ -2427,6 +2428,7 @@
   "graph": "Graph",
   "greater": "Greater Than",
   "greater_or_equals": "Greater Than or Equal",
+  "greet_user": "<1>Good {{time_of_day}},</1> <2>{{user}}</2> 👋",
```
This is fixed

yash-learner commented on public/locale/en.json:
```diff
@@ -5688,6 +5691,7 @@
   "view_all": "View All",
   "view_all_deliveries": "View All Deliveries",
   "view_all_details": "View All Details",
+  "view_analytics": "View and manage Analytics",
```
<img width="3115" height="772" alt="image" src="https://github.com/user-attachments/assets/e70a34f7-6204-4735-b87f-e34d1124afb8" />

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +50,52 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    if (hour < 12) return "Morning";
+    if (hour < 18) return "Afternoon";
+    return "Evening";
+  };
```
fixed

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +50,52 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+      visible: canViewAppointments,
     },
     {
       title: t("encounters"),
       description: t("manage_facility_users"),
-      icon: Users,
+      icon: Stethoscope,
       href: `/facility/${facilityId}/encounters/patients/${careConfig.defaultEncounterType || "all"}`,
       visible: canListEncounters,
     },
```
This is also fixed

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+          />
+
+          <div className="relative z-10 p-4 md:p-8 max-w-[250px] md:max-w-[280px] lg:max-w-lg">
+            <h1 className="mb-2 font-bold text-gray-900 text-2xl wrap-break-word">
```
this is valid in v4

yash-learner committed:
Address https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r2638943326 and one more

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -128,19 +168,25 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                          {t("go_to")} {shortcut.title}
+                        </span>
+                        <ArrowUpRight className="text-gray-400 size-4" />
+                      </div>
```
https://github.com/ohcnetwork/care_fe/pull/14906/commits/34d859f1d03ba7905b8e98562bbb0a6b31b2bf02

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                {t("welcome_back_to_hospital_dashboard")}
-              </p>
-            </div>
+        <div className="relative border-3 border-white overflow-hidden text-gray-900 rounded-xl bg-[linear-gradient(to_right,#F9FAFB_0%,#F0F0E6_60%)] sm:bg-[linear-gradient(to_right,#F9FAFB_0%, #F9FAFB_70%, #F0F0E6_100%)] lg:bg-[linear-gradient(to_right,#F5F8FF_0%,#F9FAFB_50%,#F0F0E6_100%)]">
```
Not required

yash-learner commented on public/locale/en.json:
```diff
@@ -2431,6 +2432,7 @@
   "graph": "Graph",
   "greater": "Greater Than",
   "greater_or_equals": "Greater Than or Equal",
+  "greet_user": "<1>Good {{time_of_day}},</1> <2>{{user}}</2> 👋",
```
It should be

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +52,57 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    const hour = new Date().getHours();
+    if (hour < 12) return "Morning";
+    if (hour < 18) return "Afternoon";
+    return "Evening";
```
This is fixed

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +50,52 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    (plugin) => plugin.plugin === "care_analytics_fe",
+  );
+
+  const { canViewAppointments, canListEncounters } = getPermissions(
     hasPermission,
     facilityData?.permissions ?? [],
   );
```
We are now showing appointments instead of schedule

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+        <div className="relative border-3 border-white overflow-hidden text-gray-900 rounded-xl bg-[linear-gradient(to_right,#F9FAFB_0%,#F0F0E6_60%)] sm:bg-[linear-gradient(to_right,#F9FAFB_0%, #F9FAFB_70%, #F0F0E6_100%)] lg:bg-[linear-gradient(to_right,#F5F8FF_0%,#F9FAFB_50%,#F0F0E6_100%)]">
+          <div
+            className="absolute w-full h-full bg-right bg-no-repeat bg-contain rounded-lg mix-blend-darken -right-8 md:right-0"
+            style={{ backgroundImage: "url('/images/home-banner-icon.webp')" }}
```
Not required, it is only the place used

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+          />
+          <div
+            className="absolute left-0 w-full h-full bg-left bg-no-repeat bg-contain mix-blend-darken"
+            style={{ backgroundImage: "url('/images/home-banner-sun.png')" }}
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3690314727

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -97,29 +124,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+          />
+
+          <div className="relative z-10 p-4 md:p-8 max-w-[250px] md:max-w-[280px] lg:max-w-lg">
+            <h1 className="mb-2 font-bold text-gray-900 text-2xl wrap-break-word">
```
valid in v4

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -41,34 +52,57 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    [careApps],
+  );
+
+  const { canViewAppointments, canListEncounters } = getPermissions(
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3690308445

yash-learner commented on public/locale/en.json:
```diff
@@ -671,6 +671,7 @@
   "amount_received": "Amount Received",
   "amount_tendered": "Amount Tendered",
   "amount_to_be_recorded": "This is the amount you are recording.",
+  "analytics": "Analytics",
```
will get sorted during commit

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -37,9 +40,12 @@ import {
+import { useCareApps } from "@/hooks/useCareApps";
 import facilityApi from "@/types/facility/facilityApi";
 import careConfig from "@careConfig";
+import { useMemo } from "react";
```
This is auto taken care by formatter set

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -120,29 +149,42 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                {t("welcome_back_to_hospital_dashboard")}
-              </p>
-            </div>
+        <div className="relative border-3 border-white overflow-hidden text-gray-900 rounded-xl bg-[linear-gradient(to_right,#F9FAFB_0%,#F0F0E6_60%)] sm:bg-[linear-gradient(to_right,#F9FAFB_0%, #F9FAFB_70%, #F0F0E6_100%)] lg:bg-[linear-gradient(to_right,#F5F8FF_0%,#F9FAFB_50%,#F0F0E6_100%)]">
```
outdated

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -151,19 +193,25 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
                   href={shortcut.href}
                   className="block h-full transition-all duration-200 hover:ring-2 ring-primary-400 rounded-xl ring-offset-2"
```
This is fixed

yash-learner committed:
Improve playwright test

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,40 @@
+  test("should navigate to services page when services card is clicked", async ({
+    page,
+  }) => {
+    await page.getByRole("link", { name: /services/i }).click();
```
https://github.com/ohcnetwork/care_fe/pull/14906/commits/7a0b20ba3cdec15db495c9e6b946139bea8e6b4e

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,34 +65,57 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    const hour = new Date().getHours();
+    if (hour < 12) return "Morning";
+    if (hour < 18) return "Afternoon";
+    return "Evening";
```
This  is fixed

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,40 @@
+    await page.getByRole("link", { name: /encounters/i }).click();
+
+    await expect(page.url()).toContain(
+      `/facility/${facilityId}/encounters/patients/`,
```
https://github.com/ohcnetwork/care_fe/pull/14906/commits/7a0b20ba3cdec15db495c9e6b946139bea8e6b4e

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,40 @@
+  test("should navigate to services page when services card is clicked", async ({
+    page,
+  }) => {
+    await page.getByRole("link", { name: /services/i }).click();
```
https://github.com/ohcnetwork/care_fe/pull/14906/commits/7a0b20ba3cdec15db495c9e6b946139bea8e6b4e

amjithtitus09 committed:
Merge branch 'develop' into feat/14322/home-page-redesign

nihal467 reviewed (changes requested):
- look into the failing actions and test

yash-learner committed:
Merge branch 'develop' into feat/14322/home-page-redesign

NikhilA8606 commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,34 +72,63 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+        (app) => !app.isLoading && app.plugin === "care_analytics_fe",
+      ),
+    [careApps],
+  );
```
We usually don't add plugin-specific details in the core. Instead, use `REACT_CUSTOM_SHORTCUTS` to add custom dashboard links for the plugin card.

yash-learner replied:
Fixed https://github.com/ohcnetwork/care_fe/pull/14906/commits/d7e19d43656b4b9c8b4c13ff158b3b8e37e11ac9

NikhilA8606 commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +160,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                        <CardTitle className="text-lg">
+                  <Card className="flex h-full flex-col gap-[5px] overflow-hidden rounded-lg border-0 bg-white p-1 shadow-[0_1px_3px_0_rgb(0_0_0_/_0.10),0_1px_2px_0_rgb(0_0_0_/_0.06)]">
+                    <CardContent className="flex flex-1 flex-col gap-[5px] p-0">
+                      <div className="min-h-[100px] space-y-2 rounded-t rounded-b-lg bg-gray-100 px-3 py-2.5 sm:min-h-[115px] sm:space-y-3 sm:px-4 sm:py-3">
```
Use Tailwind values instead of px values

yash-learner replied:
There is no exact tailwind equivalent for `shadow-[0_1px_3px_0_rgb(0_0_0_/_0.10),0_1px_2px_0_rgb(0_0_0_/_0.06)]`

I just matched the figma design https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care---Master?node-id=29441-53460&t=e2ZZC4ZEv31GDTX1-4

<img width="637" height="770" alt="image" src="https://github.com/user-attachments/assets/50ea9bed-7f5a-4af6-baa8-dc6cfc64a740" />

Will check with @vinutv

yash-learner replied:
Should we add a shadow-base to care theme ?

vinutv replied:
@yash-learner  No, just use the `shadow-sm` Tailwind class. It almost matches the `shadow-base` value in our design system.

There used to be a `shadow` class in Tailwind that was the base shadow. With the migration to v4, Tailwind changed the default shadow renamed the old `shadow` utility to `shadow-sm`.

https://tailwindcss.com/docs/upgrade-guide#renamed-utilities

yash-learner replied:
Fixed https://github.com/ohcnetwork/care_fe/pull/14906/commits/d7e19d43656b4b9c8b4c13ff158b3b8e37e11ac9

NikhilA8606 commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +160,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                      </div>
-                      <div>
-                        <CardTitle className="text-lg">
+                  <Card className="flex h-full flex-col gap-[5px] overflow-hidden rounded-lg border-0 bg-white p-1 shadow-[0_1px_3px_0_rgb(0_0_0_/_0.10),0_1px_2px_0_rgb(0_0_0_/_0.06)]">
```
Why the shadow className so complicated here? Can you simplify it?

yash-learner replied:
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3759479290

yash-learner replied:
Fixed https://github.com/ohcnetwork/care_fe/pull/14906/commits/d7e19d43656b4b9c8b4c13ff158b3b8e37e11ac9

NikhilA8606 commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +160,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
                           {shortcut.title}
                         </CardTitle>
-                        <CardDescription className="text-gray-500">
+                        <CardDescription className="truncate text-[11px] leading-4 font-normal text-gray-700 sm:text-xs sm:leading-normal">
```
Here too use tailwind values

yash-learner replied:
Fixed https://github.com/ohcnetwork/care_fe/pull/14906/commits/d7e19d43656b4b9c8b4c13ff158b3b8e37e11ac9

NikhilA8606 commented on src/pages/Facility/overview.tsx:
```diff
@@ -173,15 +234,15 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                  <button className="p-2 rounded-lg hover:bg-gray-100 text-gray-500 hover:text-gray-700 transition-colors">
-                    <Wrench className="size-5" />
+                  <button className="p-1.5 rounded-lg hover:bg-gray-100 text-gray-500 hover:text-gray-700 transition-all opacity-0 group-hover/pinned-links:opacity-100 focus-visible:opacity-100">
+                    <Wrench className="size-4" />
```
Why don't you use shadcn Button here?

yash-learner replied:
Fixed https://github.com/ohcnetwork/care_fe/pull/14906/commits/d7e19d43656b4b9c8b4c13ff158b3b8e37e11ac9

yash-learner committed:
Merge branch 'develop' into feat/14322/home-page-redesign

yash-learner committed:
Address review comments from https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4873958810 and bring in new Icons

yash-learner committed:
Merge branch 'develop' into feat/14322/home-page-redesign

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
+    .locator("xpath=following-sibling::div[1]");
+}
```
We do not have test-id, we do not follow that and there are no other better option so 

I need to scope clicks to that section’s grid, not “any link on the page.”

The grid is a plain <div className="grid …">. It has:

no accessible name
no role like region / list
no data-testid

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -173,15 +234,15 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                    <Wrench className="size-4" />
                   </button>
                 </DropdownMenuTrigger>
                 <DropdownMenuContent align="end">
```
addressed

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,34 +72,63 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
     hasPermission,
     facilityData?.permissions ?? [],
   );
 
```
Fixed

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +121,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                      <div className="min-h-25 space-y-2 rounded-t rounded-b-lg bg-gray-100 px-3 py-2.5 sm:min-h-28 sm:space-y-3 sm:px-4 sm:py-3">
+                        <div className="w-fit">
+                          <shortcut.icon className="size-7 text-primary sm:size-8" />
+                        </div>
```
The descriptions are meant to be short and truncate happens only on small devices like pixel 4

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+  }) => {
+    await quickLinksGrid(page)
+      .getByRole("link", { name: /go to services/i })
+      .click();
```
That is expected Bro

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+
+function quickLinksGrid(page: Page): Locator {
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3768298859

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -106,7 +110,6 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
   const customDashboardLinks = processCustomDashboardLinks(
     careConfig.customShortcuts,
     context,
```
Not scope of this ticket, this is redesign and mostly admin set those links

yash-learner commented on src/Utils/dashboardLinks.ts:
```diff
@@ -1,10 +1,60 @@
+export function isDashboardShortcutIconName(
+  name: string,
+): name is DashboardShortcutIconName {
+  return name in DASHBOARD_SHORTCUT_ICONS;
```
Fixed 

but this is 2026 and not 2025

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,44 +58,49 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+    const hour = new Date().getHours();
+    if (hour < 12) return t("morning");
+    if (hour < 18) return t("afternoon");
+    return t("evening");
```
😂

The the worth fixing this

yash-learner committed:
Address some of the review comments from https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4918313155

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                  user: formatName(user),
+                }}
+                components={{
+                  1: <span />,
```
> r at least admit it is decoration.

yep

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,44 +58,46 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
 
+  const hour = new Date().getHours();
+  const greeting =
+    hour < 12 ? t("morning") : hour < 18 ? t("afternoon") : t("evening");
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769073529

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+
+function quickLinksGrid(page: Page): Locator {
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
```
I have already replied to this duplicate review comment

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
-                      </div>
-                      <div>
-                        <CardTitle className="text-lg">
+                  <Card className="flex h-full flex-col gap-1.5 overflow-hidden rounded-lg border-0 bg-white p-1 shadow-sm">
```
its fine, it is reading what is present and not a must fix

yash-learner commented on src/Utils/dashboardLinks.ts:
```diff
@@ -1,10 +1,56 @@
+  Database: DatabaseDuoIcon,
+} as const satisfies Record<string, DashboardShortcutIcon>;
+
+export type DashboardShortcutIconName = keyof typeof DASHBOARD_SHORTCUT_ICONS;
```
We have runtime validation and compile time is a better DX and this ticket is mostly redesign so not touching it

yash-learner committed:
Address review comments from  https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4919851706

yash-learner committed:
As per discussion with Vinu use calendar with tick icon for appointments

nihal467 reviewed (changes requested):
the test is failing, look into it

yash-learner commented:
> the test is failing, look into it

Unrelated to the PR, probably flaky 

http://github.com/ohcnetwork/care_fe/actions/runs/31640201354/job/94261368909?pr=14906

```
  2 failed
    [chromium] › tests/organization/patient/encounter/organizationEncounterAccess.spec.ts:8:7 › Patient Encounter Access via Organization › can view encounter from organization patient encounters tab 
    [chromium] › tests/organization/patient/encounter/patientInfoHoverCard.spec.ts:8:7 › PatientInfoHoverCard Conditional Rendering › should NOT show Patient Home button in encounter accessed via organization route 
```

will re-run it

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+    page,
+  }) => {
+    const pinnedTitle = "Admin Availability";
+    const pinnedPath = `/facility/${facilityId}/users/admin/availability`;
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3768367101

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+function quickLinksGrid(page: Page): Locator {
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
+    .locator("xpath=following-sibling::div[1]");
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769420300

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
                           {shortcut.title}
                         </CardTitle>
-                        <CardDescription className="text-gray-500">
+                        <CardDescription className="truncate text-xs leading-4 font-normal text-gray-700 sm:leading-normal">
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3768315075

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,44 +58,46 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
     facilityData?.permissions ?? [],
   );
 
+  const hour = new Date().getHours();
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769073529

yash-learner commented on src/Utils/dashboardLinks.ts:
```diff
@@ -1,10 +1,56 @@
   description: string;
   href: string;
-  icon?: string; // Icon name from lucide-react
+  icon?: DashboardShortcutIconName;
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3770109269

yash-learner commented on public/locale/ml.json:
```diff
@@ -2300,7 +2301,6 @@
   "make_facility_public_description": "ഇത് പ്രവർത്തനക്ഷമമാക്കിയാൽ, ഈ സൗകര്യം പൊതുജനങ്ങൾക്ക് ദൃശ്യമാകും, പ്ലാറ്റ്‌ഫോം ഉപയോഗിക്കുന്ന എല്ലാവർക്കും ഇത് കണ്ടെത്താനാകും",
   "manage_and_view_questionnaires": "ക്വസ്റ്റണയർ നിയന്ത്രിക്കുകയും കാണുകയും ചെയ്യുക",
-  "manage_facility_users": "എൻകൗണ്ടർ നിയന്ത്രിക്കുക",
   "manage_my_schedule": "എന്റെ ഷെഡ്യൂൾ നിയന്ത്രിക്കുക",
```
No worries, will be taken care by crowdin updates

yash-learner commented:
> the test is failing, look into it

@nihal467 

Yeah, you're right its because of changes in the PR, I overlooked it, https://github.com/ohcnetwork/care_fe/pull/14906#issuecomment-5328837732



Fixed it https://github.com/ohcnetwork/care_fe/pull/14906/commits/9e5bd4254ad643c5663ffb9eb3feeb9477b91afa

yash-learner commented on public/locale/ml.json:
```diff
@@ -2300,7 +2301,6 @@
   "make_facility_public_description": "ഇത് പ്രവർത്തനക്ഷമമാക്കിയാൽ, ഈ സൗകര്യം പൊതുജനങ്ങൾക്ക് ദൃശ്യമാകും, പ്ലാറ്റ്‌ഫോം ഉപയോഗിക്കുന്ന എല്ലാവർക്കും ഇത് കണ്ടെത്താനാകും",
   "manage_and_view_questionnaires": "ക്വസ്റ്റണയർ നിയന്ത്രിക്കുകയും കാണുകയും ചെയ്യുക",
-  "manage_facility_users": "എൻകൗണ്ടർ നിയന്ത്രിക്കുക",
   "manage_my_schedule": "എന്റെ ഷെഡ്യൂൾ നിയന്ത്രിക്കുക",
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3805082169

yash-learner commented on public/locale/ml.json:
```diff
@@ -2300,7 +2301,6 @@
   "make_facility_public_description": "ഇത് പ്രവർത്തനക്ഷമമാക്കിയാൽ, ഈ സൗകര്യം പൊതുജനങ്ങൾക്ക് ദൃശ്യമാകും, പ്ലാറ്റ്‌ഫോം ഉപയോഗിക്കുന്ന എല്ലാവർക്കും ഇത് കണ്ടെത്താനാകും",
   "manage_and_view_questionnaires": "ക്വസ്റ്റണയർ നിയന്ത്രിക്കുകയും കാണുകയും ചെയ്യുക",
-  "manage_facility_users": "എൻകൗണ്ടർ നിയന്ത്രിക്കുക",
   "manage_my_schedule": "എന്റെ ഷെഡ്യൂൾ നിയന്ത്രിക്കുക",
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3805082169

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,44 +58,46 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
 
+  const hour = new Date().getHours();
+  const greeting =
+    hour < 12 ? t("morning") : hour < 18 ? t("afternoon") : t("evening");
```
Addressed valid part of the review comment https://github.com/ohcnetwork/care_fe/pull/14906/commits/c95f6dc889934749bd6ee847a0f89e7d858df0ae

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -59,44 +58,46 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
 
+  const hour = new Date().getHours();
+  const greeting =
+    hour < 12 ? t("morning") : hour < 18 ? t("afternoon") : t("evening");
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3804898570

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+    page,
+  }) => {
+    const pinnedTitle = "Admin Availability";
+    const pinnedPath = `/facility/${facilityId}/users/admin/availability`;
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769420300

yash-learner commented on src/Utils/dashboardLinks.ts:
```diff
@@ -1,10 +1,56 @@
   description: string;
   href: string;
-  icon?: string; // Icon name from lucide-react
+  icon?: DashboardShortcutIconName;
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3770109269

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
+    .locator("xpath=following-sibling::div[1]");
+}
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769420300

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
                           {shortcut.title}
                         </CardTitle>
-                        <CardDescription className="text-gray-500">
+                        <CardDescription className="truncate text-xs leading-4 font-normal text-gray-700 sm:leading-normal">
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3768315075

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                  <Card className="flex h-full flex-col gap-1.5 overflow-hidden rounded-lg border-0 bg-white p-1 shadow-sm">
+                    <CardContent className="flex flex-1 flex-col gap-1.5 p-0">
+                      <div className="min-h-25 space-y-2 rounded-t rounded-b-lg bg-gray-100 px-3 py-2.5 sm:min-h-28 sm:space-y-3 sm:px-4 sm:py-3">
+                        <div className="w-fit">
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3768315075

yash-learner commented on src/Utils/dashboardLinks.ts:
```diff
@@ -1,10 +1,56 @@
+  }
+  return DEFAULT_DASHBOARD_SHORTCUT_ICON;
+}
 
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3770109269

yash-learner commented on src/pages/Facility/overview.tsx:
```diff
@@ -118,52 +118,71 @@ export function FacilityOverview({ facilityId }: FacilityOverviewProps) {
+                  time_of_day: greeting,
+                  user: formatName(user),
+                }}
+                components={{
```
That is a design/product change -> if they still want they use pinned links or add via env

yash-learner commented on tests/facility/overview/overview.spec.ts:
```diff
@@ -0,0 +1,108 @@
+
+function quickLinksGrid(page: Page): Locator {
+  return page
+    .getByRole("heading", { name: /^quick links$/i })
```
https://github.com/ohcnetwork/care_fe/pull/14906#discussion_r3769420300

yash-learner commented:
I have cancelled the last run of grumpy reviewer I resolved all the review comments and made fixes for relevant ones

and it still generated same duplicate comments across last three runs https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4961579020, https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4961909844, https://github.com/ohcnetwork/care_fe/pull/14906#pullrequestreview-4962055358

it feels enervating and  onerous

rithviknishad reviewed (approved):
LGTM
